pkg/sidecar/pod.go

```diff
@@ -81,5 +102,12 @@ func existsIn(pod corev1.Pod) bool {
 			return true
 		}
 	}
+	// NOTE: we also check init containers (native sidecars) since k8s 1.28.
+	// This should have no side effects.
+	for _, container := range pod.Spec.InitContainers {
+		if container.Name == naming.Container() {
+			return true
+		}
+	}
```

Comment:
How about https://pkg.go.dev/slices#ContainsFunc? Both here and above. Also, shouldn't this only happen if the feature flag is set?